pilot/pkg/networking/core/v1alpha3/route/route_test.go

```diff
@@ -523,7 +513,7 @@ var networkingDestinationRuleWithPortLevelTrafficPolicy = &networking.Destinatio
 					LbPolicy: loadBalancerPolicy("hash-cookie-1"),
 				},
 				Port: &networking.PortSelector{
-					Port: &networking.PortSelector_Number{Number: 8080},
+					Number: 8080,
```

Comment:
Is this backwards compatible? Or is the yaml the same and just golang is different?

Reply:
@howardjohn I'm not sure how it works, because in the istio/api and go code it definitely looks like it goes from `Port.Port.Number` to `Port.Number` [here](https://github.com/istio/api/pull/1100/files?file-filters%5B%5D=.html&file-filters%5B%5D=.json&file-filters%5B%5D=.proto&file-filters%5B%5D=.py#diff-0820bb0406265ba9b9a9e11dcec06601)
but I did not have to modify any of the e2e VirtualService files which use PortSelector after pulling in the change:
https://github.com/istio/istio/blob/master/tests/e2e/tests/pilot/testdata/networking/v1alpha3/virtualservice-route-tcp-weighted.yaml#L15-L19

Reply:
https://developers.google.com/protocol-buffers/docs/proto3#backwards-compatibility-issues suggests that `oneof` removal _should_ be backwards compatible in this use case.